include/aikido/control/BarrettHandPositionCommandExecutor.hpp

```diff
@@ -1,86 +1,28 @@
-/// and distal joint moves with certain mimic ratio until collision.
-/// If 
+/// Abstract class for position commands for a Barrett Hand.
 class BarrettHandPositionCommandExecutor
 {
 public:
-  /// Constructor.
-  /// \param _positionCommandExecutors 3 executors to control
-  ///        proximal and distal joints of the fingers. The first two fingers
-  ///        should have (spread, proximal, distal) joints. 
-  ///        Third finger should have 2 joints (proximal, distal).
-  /// \param _spreadCommandExecutor Executors to control
-  ///        spreads of the fingers.
-   BarrettHandPositionCommandExecutor(
-    std::array<BarrettFingerPositionCommandExecutorPtr, 3> _positionCommandExecutors,
-    BarrettFingerSpreadCommandExecutorPtr _spreadCommandExecutor);
+  virtual ~BarrettHandPositionCommandExecutor() = default;
 
-  /// Set relevant variables for moving fingers.
-  /// In order to move the fingers, step method should be called multiple times 
-  /// until future returns.
-  /// \param _goalPositions End dof pose for proximal joints and spread. 
-  ///        First 3 should be for proximal joints, the last element should be
-  ///        for spread. If _positions are above/below joint limits,
-  ///        the fingers will move only upto the limit.
-  /// \param _collideWith CollisionGroup to check collision with fingers.
+  /// Execute hand to goalPosition
+  /// \param goalPositions End dof pose for proximal joints and spread.
   /// \return Future which becomes available when the execution completes.
-  std::future<void> execute(
-    Eigen::Matrix<double, 4, 1> _goalPositions, 
-    ::dart::collision::CollisionGroupPtr _collideWith);
+  virtual std::future<void> execute(
+    Eigen::Matrix<double, 4, 1> goalPositions) = 0;
 
-  /// Calls step method of finger executors.
-  /// If multiple threads are accessing this function or skeleton associated
-  /// with this executor, it is necessary to lock the skeleton before
-  /// calling this method.
-  /// \param _timeSincePreviousCall Time since previous call. 
-  void step(double _timeSincePreviousCall);
-
-private:
-
-  constexpr static int kNumPositionExecutor = 3;
-  constexpr static int kNumSpreadExecutor = 1; 
-  constexpr static auto kWaitPeriod = std::chrono::milliseconds(1);
-
-  /// Executor for proximal and distal joints. 
-  std::array<BarrettFingerPositionCommandExecutorPtr, 3> mPositionCommandExecutors;
-  BarrettFingerSpreadCommandExecutorPtr mSpreadCommandExecutor;
-
-  std::unique_ptr<std::promise<void>> mPromise;
-
-  std::vector<std::future<void>> mFingerFutures;
-
-  /// Control access to mPromise, mInExecution
-  /// mProximalGoalPositions, mSpreadGoalPositin, mSpread
-  std::mutex mMutex;
-
-  /// Flag for indicating execution of a command. 
-  bool mInExecution;
-
-  /// Values for executing a position and spread command. 
-  Eigen::Vector3d mProximalGoalPositions; 
-  double mSpreadGoalPosition;
+  virtual void step(double timeSincePreviousCall) = 0;
```

Comment:
Remove the `timeSincePreviousCall` argument. If this is necessary inside some executors, maintain this state as a member variable.